### DISCUSSION TITLE
Correct approach for extending e.g. the DatePicker module

### DISCUSSION THREAD

**Discussion by ghost** [vmware/clarity#5694] [Q&A]
Hi all,

I hope someone can help me. I am currently involved in evaluating clarity.design for use in an Angular app. However, an absolute necessity for us is the possibility of being able to (simply) modify some aspects of the behavior and html structure of certain components.

For example: in the **ClrCalendar** component (ClrDatepicker module), we would need to add _weeknumbers_ and a "_today_" button. However, being relatively new also in Angular I am currently struggling a bit so to say. One approach I was trying was to exploit @Directives, but these do not allow me to change a components templates.

However, is my goal achievable in a sane way?

**Comment by hippee-lee** [comment]
For the Today button is this not doing what you need it to do? 
![Screen Shot 2021-03-08 at 10 26 21 AM](https://user-images.githubusercontent.com/433692/110364769-0b69cf00-7ff9-11eb-9cec-ccee54e412fc.png)

I'm not quite sure what is meant by `weeknumbers` can you elaborate?

I do not think it is possible or a good idea to modify templates at runtime. However, could the app wrap what it needs inside a component you control in a way that adds the additional UI / info?

**Reply by ghost** [reply]
Thank you for your answer! 

The "current month" button is indeed _similar_, but not exactly what I am after: what I would like to implement is a button that sets the datepicker value to todays date – immediately closing the overlay. 

As for the weeknumbers, I have attached a screenshot: 
![clarity-picker](https://user-images.githubusercontent.com/78234161/110394839-5c1b0f80-806d-11eb-9436-1e62cf8ce988.jpg)

You are absolutely right regarding "not a good idea to modify templates at runtime". 
However, what I tried was:
- Define a component that "overwrites" the tag used by the **ClrCalendar** class ("clr-calendar", I think it was) and adjusting the template in the process; however, I guess Angular does not allow to do such things (which is plausible); 
- I managed to define a directive for the main tag ("clr-date-container"), however, there seems no good way of refine some aspects of the template; 
- Finally, as a last resort, I tried to extend **ClrDateContainer** – however, I soon ran into issues with private services. 

The thing is, we are a bit between a rock and a hard place. :) Unfortunately we need to frequently modify components – but we do not have the resources to write our own. In the same time, I would like to keep all changes easily traceable and manageable, making it easy for all colleagues to work with an already fine library!
